modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2922,6 +2924,7 @@ def _compute_duplicated(df):  # pragma: no cover
     # return a new one from here and let the front end handle the inplace
     # update.
     def insert(self, loc, column, value):
+        value = self._wrap_column_data(value)
```

Comment:
Looks fine as a temporary hack.

However, there's a TODO message a few lines below mentioning that the insertion of a non-distributed data can be speeded-up significantly by changing `apply_full_axis...` call to the `apply_select_indices(item_to_distribute=value, ...)`

https://github.com/modin-project/modin/blob/781fed8152a6e0a6fcae9cab5604644577103a5f/modin/core/storage_formats/pandas/query_compiler.py#L2945-L2946

This way, we wouldn't need to wrap the newly created dataframe but propagate the value directly to the partitions, which, in theory, should be quite fast.

Could you please quickly check the approach from TODO message makes sense? If it'll be too inefficient or too complex to implement, we can stick to the changes introduced by this PR